cborg/src/Codec/CBOR/Decoding.hs

```diff
@@ -769,7 +780,13 @@ decodeTag64Canonical :: Decoder s Word64
 {-# INLINE decodeTag64Canonical #-}
 decodeTag64Canonical =
 #if defined(ARCH_64bit)
-  Decoder (\k -> return (ConsumeTagCanonical (\w# -> k (W64# w#))))
+  Decoder (\k -> return (ConsumeTagCanonical (\w# -> k (W64#
+#if MIN_VERSION_base(4,17,0)
+    (wordToWord64# w#)
+#else
+    w#
+#endif
+  ))))
```

Comment:
Can this be a CPP macro instead? Using #if’s everywhere feels messy to me.

Reply:
This is the limit of my CPP knowledge - I'd love it if there was an easier way!